### PR TITLE
Tune MaxLengthPlugin behavior for restoration

### PR DESCRIPTION
**Pull request by trueadm**
Based off further feedback from https://github.com/facebook/lexical/issues/2230#issuecomment-1137838765. We can actually apply a heuristic where if the last editor state content was already at the max length, we just try and restore that editor state entirely.

### CODE REVIEW

File: packages/lexical-utils/src/index.ts [hunk 1]
```diff
@@ -382,3 +388,17 @@ export function unstable_convertLegacyJSONEditorState(
...
+
+export function $restoreEditorState(
+  editor: LexicalEditor,
+  editorState: EditorState,
+): void {
+  const nodeMap = new Map(editorState._nodeMap);
+  const activeEditorState = editor._pendingEditorState;
+  activeEditorState._nodeMap = nodeMap;
+  for (const [, node] of nodeMap) {
+    node.markDirty();
```
Comment (fantactuka) [inline packages/lexical-utils/src/index.ts:400]: Maybe `editor._dirtyType = FULL_RECONCILE`?

**Review by fantactuka — APPROVED** [review]

File: packages/lexical-utils/src/index.ts (hunk 1 above)
Comment (trueadm) [reply packages/lexical-utils/src/index.ts:400]: I thought that, but then won't other node transforms not run?

Comment (fantactuka) [reply packages/lexical-utils/src/index.ts:400]: Didn't know that's the difference! But if it's a restored state, then it should already have all transformers resolved (similar as in history stack we don't want any transformers when undo/redo)

Comment (trueadm) [reply packages/lexical-utils/src/index.ts:400]: The big difference is that is done via `setEditorState`. I guess it's fine to make that change though, we just need to keep on eye on this.

**Pull request merged**